charts/hlf-k8s/templates/deployment-chaincode-operator.yaml

```diff
@@ -40,48 +41,103 @@ spec:
       serviceAccountName: {{ template "substra.fullname" $ }}
       containers:
         - name: fabric-tools
-          image: {{ $.Values.image.repository }}:{{ $.Values.image.tag }}
+          image: {{ $.Values.images.hlfk8s.repository }}:{{ $.Values.images.hlfk8s.tag }}
           command: ['bash', '-c']
           args:
             - |
               ## Update CA certs
               update-ca-certificates
 
+              ## Check connection with the Peer
+              printf "[DEBUG] Testing the connection with the Peer ({{ index $.Values "hlf-peer" "host" }}:{{ index $.Values "hlf-peer" "port" }})\n"
+              until $(nc -z {{ index $.Values "hlf-peer" "host" }} {{ index $.Values "hlf-peer" "port" }}); do
+                printf "[DEBUG] Peer ({{ index $.Values "hlf-peer" "host" }}:{{ index $.Values "hlf-peer" "port" }}) is not reacheable, retry in 5s\n"
+                sleep 5
```

Comment:
Sleep 5 seems good as waiting only for 1 second is maybe too short

Reply:
Why too short?

Reply:
I don't know what is the good time to sleep. But it will take to much time to do a benchmark to know what is the good value for the sleep, and this value of 5 is used everywhere we wait for a service like a peer or an orderer :)

Reply:
I measured current v.s. "without sleep" and here are the results:

```
# Actuel: avec `sleep X` (X > 0) 
real    0m30.736s
real    0m34.121s
real    0m32.005s
# Test: avec `sleep 0.1`
real    0m20.118s
real    0m28.689s
real    0m30.416s
real    0m25.137s 
```

There's a difference, but it's overall minor.

Reply:
I measured the deployment stabilization time + the time for the chaincode to be ready in the "current" v.s. "without sleep" scenarios. 

Here are the results:

```
# Current: with `sleep X` (X > 0) 
Deployments stabilized in 26.899945242s
real    0m3.097s
Deployments stabilized in 26.560970816s
real    0m3.188s
Deployments stabilized in 28.159576478s
real    0m3.145s
Deployments stabilized in 24.25722498s
real    0m3.176s
Deployments stabilized in 25.591797322s
real    0m3.230s

# Test: with `sleep 0.1` (I replaced all the sleep X with sleep 0.1 in the code)
Deployments stabilized in 2m23.142196021s
real    0m5.069s
Deployments stabilized in 15.300134681s
real    0m4.595s
Deployments stabilized in 9.724714045s
real    0m4.274s
Deployments stabilized in 18.236987181s
real    0m4.642s
Deployments stabilized in 8.295457046s
real    0m4.399s
```

Test code: 

```
for i in `seq 4`; do 
    ./examples/dev-secrets.sh create
    sk run
    { time while true; do ./examples/test-dev-network.sh && break;done } 2>&1 | tee -a with-sleep.txt
    sk delete
done
````

Reply:
I see `2m23.142196021s`, looks big no ?
What is the conclusion of your test ?

Reply:
Yeah that's an outlier, I'm not sure what happened there.

> What is the conclusion of your test ?

Excluding the outlier, the deployment is 10-15 sec faster without the sleeps. It's just a data point to have an idea of how `sleep`s affect deployment time.

Reply:
Yes, I think we can do this optimisation in another PR :)